### PR TITLE
Azure Monitor : Query more than 10 dimensions ( Fixes #17230 )

### LINKED ISSUE #17230: AzureMonitor: Limited to top 10 Instances for a given metric
If a given metric has more than 10 unique instances, only 10 are returned, since the &$top={$top} parameter is not configurable.

### PR DESCRIPTION
**Pull request by yesoreyeram**
**What this PR does / why we need it**:

Query more than 10 dimensions in Azure Monitor API. Azure monitor pulls only 10 dimensions by default.  This PR gives ability to pull more than 10 dimension instances.

image

Fixes #17230

### CODE REVIEW

**Comment by marefr** [comment]
@yesoreyeram Could you please merge with/rebase on latest master and we'll have a look at this. Thanks

**Comment by yesoreyeram** [comment]
@marefr  - I dont see any merge conflicts. Still do you want me to rebase?

> @yesoreyeram Could you please merge with/rebase on latest master and we'll have a look at this. Thanks

**Comment by marefr** [comment]
@yesoreyeram we reverted a change in master that's probably why :)

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts [hunk 1]
```diff
@@ -75,6 +75,7 @@ export default class AzureMonitorDatasource {
       const metricDefinition = this.templateSrv.replace(item.metricDefinition, options.scopedVars);
       const timeGrain = this.templateSrv.replace((item.timeGrain || '').toString(), options.scopedVars);
       const aggregation = this.templateSrv.replace(item.aggregation, options.scopedVars);
+      const top = this.templateSrv.replace(item.top || `10`, options.scopedVars);
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts]: replace item.top || \`10\` with item.top || ''
you never want to intercept 10 as a template variable

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts [hunk 2]
```diff
@@ -95,6 +96,7 @@ export default class AzureMonitorDatasource {
             metricNamespace && metricNamespace !== this.defaultDropdownValue ? metricNamespace : metricDefinition,
           aggregation: aggregation,
           dimension: this.templateSrv.replace(item.dimension, options.scopedVars),
+          top: top || `10`,
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_monitor/azure_monitor_datasource.ts]: replace backticks with single quotes

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts [hunk 3]
```diff
@@ -71,6 +72,7 @@ export class AzureMonitorQueryCtrl extends QueryCtrl {
       metricName: this.defaultDropdownValue,
       dimensionFilter: '*',
       timeGrain: 'auto',
+      top: `10`,
```
Comment (sunker) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/query_ctrl.ts]: single quotes

**Review by sunker — CHANGES_REQUESTED** [review]
Great work! Just fix the minor changes that was requested and I'm ready to merge.

**Review by sunker — APPROVED** [review]

**Pull request merged**